lib/moonshot/stack.rb

```diff
@@ -195,6 +198,19 @@ def load_parameters_file
       end
     end
 
+    def load_template_file
+      json_template = JsonStackTemplate.new(template_file, log: @log)
```

Comment:
Rather than depending on a JSON parser error in the YAML file, I think you want to do something like:
```ruby
json_template = JsonStackTemplate.new(json_template_file, log: @log)
yaml_template = YamlStackTemplate.new(yaml_template_file, log: @log)
```

In this way, the `#exists?` method can decide if the class found a file or not. This would, of course, mean lazy-loading the file only when necessary, rather than in the constructor, which might be ideal anyway.